R/apply_statistic_fmt_fn.R

```diff
@@ -112,7 +112,7 @@ alias_as_fmt_fn <- function(x, call = parent.frame()) {
 label_cards <- function(digits = 1, scale = 1, width = NULL) {
   function(x) {
     # round and scale vector
-    res <- format(round5(x * scale, digits = digits), nsmall = digits)
+    res <- format(round5(x * scale, digits = digits), nsmall = digits) |> str_trim()
```

Comment:
When we added significative digits, we ended up using formatC for this. Do you think it could be a better fit/function?

Reply:
I couldn't tell you the difference between the two. What is the improvement on `formatC()`? Thanks!

Reply:
Took a quick look, and it doesn't look like it adds trailing zeros.  But there must be some way to get it working. But I guess I need to still understand how `formatC()` is a better solution 🤔 

```r
> formatC(1:10 / 10, digits = 2)
 [1] "0.1" "0.2" "0.3" "0.4" "0.5" "0.6" "0.7" "0.8" "0.9" "  1"
```

Reply:
I am not sure it is actually better but I found the following:

If you need precise control over the formatting, including specifying the number of significant digits, padding, alignment, or different numeric formats (like fixed or scientific notation), formatC is the better choice.
If your primary goal is simply to round numbers to a certain number of decimal places and then convert them to a more readable string format without needing detailed control over the formatting, then using format(round(.)) might be simpler and more straightforward.

We use: `str_vals <- formatC(x, digits = detect_dig, format = "f")` for a direct sig. digit and the following for sig digits (it should be semplified imo)
```r
format_sigfig <- function(sigfig, format = "xx", num_fmt = "fg") {
  checkmate::assert_integerish(sigfig)
  format <- gsub("xx\\.|xx\\.x+", "xx", format)
  checkmate::assert_choice(format, c("xx", "xx / xx", "(xx, xx)", "xx - xx", "xx (xx)"))
  function(x, ...) {
    if (!is.numeric(x)) stop("`format_sigfig` cannot be used for non-numeric values. Please choose another format.")
    num <- formatC(signif(x, digits = sigfig), digits = sigfig, format = num_fmt, flag = "#")
    num <- gsub("\\.$", "", num) # remove trailing "."

    format_value(num, format)
  }
}
```

In general, whatever works for me! I was just curious if you had a strong opinion about using `format(round5())`

Reply:
Ah, thank you for the details. We do need to use `round5()` because it implements regular human rounding, and not R's default "round-to-even"